src/libs/SearchUtils.ts

```diff
@@ -299,34 +300,32 @@ function getSortedReportData(data: ReportListItemType[]) {
     });
 }
 
-function getSearchParams() {
+function getCurrentSearchParams() {
     const topmostCentralPaneRoute = getTopmostCentralPaneRoute(navigationRef.getRootState() as State<RootStackParamList>);
     return topmostCentralPaneRoute?.params as AuthScreensParamList['Search_Central_Pane'];
 }
 
+// Query may be in the q or cq parameter
+function getQueryStringFromParams(params: AuthScreensParamList[typeof SCREENS.SEARCH.CENTRAL_PANE]) {
+    return params.q ?? params.cq;
+}
+
+function isCustomQueryFromParams(params: AuthScreensParamList[typeof SCREENS.SEARCH.CENTRAL_PANE]) {
+    return !!params.cq;
+}
+
 function isSearchResultsEmpty(searchResults: SearchResults) {
     return !Object.keys(searchResults?.data).some((key) => key.startsWith(ONYXKEYS.COLLECTION.TRANSACTION));
 }
 
-function getQueryHashFromString(query: string): number {
+function getQueryHashFromString(query: SearchQueryString): number {
     return UserUtils.hashText(query, 2 ** 32);
 }
 
-type JSONQuery = {
-    input: string;
-    hash: number;
-    type: string;
-    status: string;
-    sortBy: string;
-    sortOrder: string;
-    offset: number;
-    filters: ASTNode;
-};
-
-function buildJSONQuery(query: string) {
+function buildSearchQueryJSON(query: SearchQueryString) {
```

Comment:
Why did we rename this function?

Reply:
There is a story to this change. It's not a strong argument but I believe it makes sense 😄

If the function is named `buildJSONQuery` then the result will look like that:

```ts
const JSONQuery = buildJSONQuery()
```
 right?

But then how we should name the return type of `builJSONQuery` function? it should start with the capital letter so `JSONQuery` but it is taken.

We could name the result differently so 

```ts
const jsonQuery = buildJSONQuery()
```

But it felt weird to me to change the whole acronym to small or big caps depending on the context.

And what if we want to add prefix? e.g. `current?`

It should be:
`currentJSONQuery` OR `currentJsonQuery`?

Then I realized that if we can have `queryString` then `queryJSON` and `queryAST` also should work for a convention. 

So finally I chose the easy way which is avoiding acronym at the beginning of the name 😄

Reply:
ahah I don't see an issue with the naming, but also this is not a big deal. Let's keep it as you have here already